### PR TITLE
Update Webpack to fix elliptic 6.5.3 vulnerability issue

### PR DESCRIPTION
**Pull request by igor-ribeiiro**
Elliptic 6.5.3 has a vulnerability issue and needs to be removed from the code. It currently is imported by ```webpack``` in package ```custom-functions-metadata-plugin```. Thus ```webpack``` needed to be updated to a major version, and that's why there are a lot of changes.

Tests:
- Tested by running and building the excel-custom-functions project

### CODE REVIEW

**Comment by akrantz** [comment]
Does this mean that Excel-Custom-Functions must use webpack v5?  If this requires webpack 5, then I suggest the major version be bumped when published. 

You should publish to the next dist-tag, then validate by installing custom-functions-metadata-plugin@next into Excel-Custom-Functions and making sure it works.

**Comment by igor-ribeiiro** [comment]
> Does this mean that Excel-Custom-Functions must use webpack v5? If this requires webpack 5, then I suggest the major version be bumped when published.
> 
> You should publish to the next dist-tag, then validate by installing custom-functions-metadata-plugin@next into Excel-Custom-Functions and making sure it works.

I didn't even think about that, but yes, the webpack version should be updated as well in Excel-Custom-Functions, given that the current version uses a vulnerable package. Currently only the webpack version inside the plugin has been updated

**Comment by akrantz** [comment]
Please resolve merge conflicts. These changes are fine. 

I really want to understand if the consumers of CustomFunctionsMetadataPlugin in webpack config means that webpack 5 must now be used. 

We will want to do a major version bump of the custom-functions-metadata-plugin package when "npm run publish:next" is run,

**Review by akrantz — CHANGES_REQUESTED** [review]
Ping me after you merge.

**Comment by igor-ribeiiro** [comment]
> Please resolve merge conflicts. These changes are fine.
> 
> I really want to understand if the consumers of CustomFunctionsMetadataPlugin in webpack config means that webpack 5 must now be used.
> 
> We will want to do a major version bump of the custom-functions-metadata-plugin package when "npm run publish:next" is run,

I just resolved the merge conflicts.
Sure, I will use the publish with the @next tag to be extra safe.

The consumers of CustomFunctionsMetadataPlugin, like the Excel Custom Functions, can just keep using their webpack versions normally if they are dev dependencies.

**Review by millerds — DISMISSED** [review]

File: packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts [hunk 1]
```diff
@@ -27,17 +29,14 @@ class CustomFunctionsMetadataPlugin {
...
-                    source() { return generateResult.metadataJson; },
-                    size() { return generateResult.metadataJson.length; },
-                };
+                compilation.assets[this.options.output] = new sources.RawSource(generateResult.metadataJson);
             }
         });
 
         compiler.hooks.compilation.tap(pluginName, (compilation, params) => {
-            compilation.moduleTemplates.javascript.hooks.render.tap(pluginName, (source, module) => {
+            compilation.moduleTemplates.javascript.hooks.render.tap(pluginName, (source : any, module : any) => {
```
Comment (akrantz) [inline packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts:39]: I'm getting build errors for these changes...

**Review by akrantz — CHANGES_REQUESTED** [review]
This does not build locally for me. 

```
[0]
[0] > custom-functions-metadata-plugin@1.0.44 tsc
[0] > tsc -p tsconfig.json
[0]
[0] src/customfunctionsplugin.ts(7,49): error TS7016: Could not find a declaration file for module 'webpack'. 'c:/dev/git/Office-Addin-Scripts/packages/custom-functions-metadata-plugin/node_modules/webpack/lib/webpack.js' implicitly has an 'any' type.
[0]   Try `npm install @types/webpack` if it exists or add a new declaration (.d.ts) file containing `declare module 'webpack';`
[0] src/customfunctionsplugin.ts(30,46): error TS7006: Parameter 'compilation' implicitly has an 'any' type.
[0] src/customfunctionsplugin.ts(38,53): error TS7006: Parameter 'compilation' implicitly has an 'any' type.
[0] src/customfunctionsplugin.ts(38,66): error TS7006: Parameter 'params' implicitly has an 'any' type.
[0] npm ERR! code 2
[0] npm ERR! path c:\dev\git\Office-Addin-Scripts\packages\custom-functions-metadata-plugin
[0] npm ERR! command failed
[0] npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c tsc -p tsconfig.json
```

File: packages/custom-functions-metadata-plugin/package.json [hunk 2]
```diff
@@ -26,11 +26,10 @@
   "dependencies": {
     "custom-functions-metadata": "^1.0.44",
     "es6-promise": "^4.2.8",
-    "webpack": "^4.42.1"
+    "webpack": "^5.24.4"
   },
   "devDependencies": {
     "@types/node": "^12.12.34",
-    "@types/webpack": "^4.41.10",
```
Comment (akrantz) [inline packages/custom-functions-metadata-plugin/package.json:33]: I figured out that @types/webpack dev dependency is now missing.

File: packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts (hunk 1 above)
Comment (igor-ribeiiro) [reply packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts:39]: Wait, these changes seem to fix your bug!
Like, I added the ```any``` property on my own to fix this problem, so it is weird that this is happening

**Review by akrantz — APPROVED** [review]
OK, we worked through stuff so it is building now.

File: packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts (hunk 1 above)
Comment (akrantz) [reply packages/custom-functions-metadata-plugin/src/customfunctionsplugin.ts:39]: After getting the latest Node and NPM, I can now build everything.

File: packages/custom-functions-metadata-plugin/package.json (hunk 2 above)
Comment (akrantz) [reply packages/custom-functions-metadata-plugin/package.json:33]: webpack v5 no longer needs separate @types/webpack.

**Pull request merged**